### ISSUE TITLE
Overridden built-in setter functions are not called when setting a value, even when using `self`

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot#48843]
**Godot version:**

3.3 / 3.3.1

**OS/device including version:**

Windows 10

**Issue description:**

Overridden built-in setter functions are not called when setting a value, even when using the `self` keyword.
(Maybe this is expected, but it doesn't seem right. You have to always remember to call the setter.)

**Steps to reproduce:**
(See example project)

**Minimal reproduction project:**

[Method Override Bug.zip](https://github.com/godotengine/godot/files/6509348/Method.Override.Bug.zip)

**Comment by Calinou** [comment]
I think this is done by design. You can use a `_set()` function to create magic setters/getters, but keep in mind this has a performance cost.

**Comment by Error7Studios** [comment]
> You can define a _set() function to create magic setters/getters

I tried to do that, but this gives me a stack overflow.

```
func _set(property: String, value) -> bool:
	if get(property) != null:
		match property:
			"bbcode_text":
				"< code here >"
			_:
				pass
		set(property, value)
		return true
	else:
		return false
```

However, looks like this is already fixed by [#47672](https://github.com/godotengine/godot/pull/47672).
Also, according to [#844](https://github.com/godotengine/godot-proposals/issues/844), setters will get called by default in 4.0, so this will no longer be an issue.

> I think this is done by design.

In that case, I'll close this, since 4.0 is almost here and I don't think a change like this will be made for 3.x
